src/store/modules/nft.js

```diff
@@ -395,17 +390,18 @@ const actions = {
   removeNFTFiatPriceInfoByClassId({ commit }, classId) {
     commit(TYPES.NFT_SET_NFT_CLASS_FIAT_PRICE_INFO, {
       classId,
-      data: undefined,
+      fiatPrice: null,
     });
   },
   async fetchNFTFiatPriceInfoByClassId({ commit }, classId) {
     if (!classId) return undefined;
-    const { fiatPrice, listingInfo } = await this.$api.$get(
+    const { fiatPrice } = await this.$api.$get(
       api.getStripeFiatPrice({ classId })
     );
-    const data = { fiatPrice, listingInfo };
-    commit(TYPES.NFT_SET_NFT_CLASS_FIAT_PRICE_INFO, { classId, data });
-    return data;
+    if (Array.isArray(classId)) {
+      commit(TYPES.NFT_SET_NFT_CLASS_FIAT_PRICE_INFO, { classId, fiatPrice });
```

Comment:
Why check for array here but we doesn't handle it with irritated
Iterators? How about the case if it's not array?

Reply:
It's a bit tricky that the current `fiatPrice` for a single class (not array case) includes a base tx fee, but actually, we should further break it down. Could be in this PR or another PR?

Reply:
Where do we display this price in this PR?

Reply:
I was planning displaying fiat price on the button before redirecting to Stripe page.
That way we should fetch `GET /price` API when mounted.
But that will store a very long key into the store, which is not reusable, and should fetch again if any NFT class in the list is removed.

Do you think we should simply store the LIKE/USD rate and the base tx fee provided by API in store, and calculate the fiat price to display on button? Similar logic for purchasing a single class.

(Or leave it intact in this PR, abd refactor the collect modal and the shopping cart functions in some way soon?

Reply:
Just directly fetch the total fiat price once when we enter shopping cart page, don't calculate using like/usd

Reply:
What about if the user removes some classes in the shopping cart?

Reply:
Refetch from the api

Reply:
I think for this we should keep it as is old behaviour, and ignore array for now
so the guard would actually be `if (!Array.isArray(classId)) {`

Reply:
Yes, a typo